architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,693 @@
+to proceed with coding and PRs.
+
+If there are alternatives you are considering please include that in the open
+questions section. If the product has a layered architecture, it's good to
+align these sections with the product's layers. This will help readers use
+their current understanding to understand your ideas.
+
+Discuss the rationale behind architectural choices and alternative options 
+considered during the design process.
+-->
+
+#### BicepDeployment Custom Resource Definition
+
+The `BicepDeployment` CRD will be a new Kubernetes CRD that will be used to contain the ARM JSON template and parameters for deploying Radius resources. The CRD will have the following fields:
+
+```go
+// BicepDeploymentSpec defines the desired state of a BicepDeployment
+type BicepDeploymentSpec struct {
+  // Template is the ARM JSON template that defines the resources to deploy.
+  Template string `json:"template"`
+
+  // Parameters is the ARM JSON parameters file that specifies the parameters for the template.
+  Parameters string `json:"parameters"`
+}
+
+// BicepDeploymentStatus defines the observed state of the Bicep Deployment.
+type BicepDeploymentStatus struct {
+  // ObservedGeneration is the most recent generation observed for this BicepDeployment.
+  ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Scope is the resource ID of the scope.
+	Scope string `json:"scope,omitempty"`
+
+	// Resource is the resource ID of the deployment.
+	Resource string `json:"resource,omitempty"`
+
+	// Operation tracks the status of an in-progress provisioning operation.
+	Operation *ResourceOperation `json:"operation,omitempty"`
+
+	// Phrase indicates the current status of the Bicep Deployment.
+	Phrase BicepDeploymentPhrase `json:"phrase,omitempty"`
+
+  // Description is a human-readable description of the status of the Bicep Deployment.
+  Description string `json:"description,omitempty"`
+
+  // Resources is a list of resources that are created by the BicepDeployment.
+  Resources []BicepDeploymentResourceRef `json:"resources,omitempty"`
+}
+
+// BicepDeploymentPhrase is a string representation of the current status of a Bicep Deployment.
+type BicepDeploymentPhrase string
+
+const (
+	// BicepDeploymentPhraseUpdating indicates that the Bicep Deployment is being updated.
+	BicepDeploymentPhraseUpdating BicepDeploymentPhrase = "Updating"
+
+	// BicepDeploymentPhraseReady indicates that the Bicep Deployment is ready.
+	BicepDeploymentPhraseReady BicepDeploymentPhrase = "Ready"
+
+	// BicepDeploymentPhraseFailed indicates that the Bicep Deployment has failed.
+	BicepDeploymentPhraseFailed BicepDeploymentPhrase = "Failed"
+
+	// BicepDeploymentPhraseDeleting indicates that the Bicep Deployment is being deleted.
+	BicepDeploymentPhraseDeleting BicepDeploymentPhrase = "Deleting"
+
+	// BicepDeploymentPhraseDeleted indicates that the Bicep Deployment has been deleted.
+	BicepDeploymentPhraseDeleted BicepDeploymentPhrase = "Deleted"
```

Comment:
How long will we keep the resource around, i.e. as 'deleted'? Is clean up implicit or explicit?